### PR TITLE
[Feature] Admin Registration/Passwords Refactor

### PR DESCRIPTION
**Pull request by Ghalko**
## Purpose
Finish #4992
## Changes

Cleaning up and conforming to Rest of Admin module

## Ticket
ADM-41, ADM-47

### CODE REVIEW

File: admin/common_auth/admin.py [hunk 1]
```diff
@@ -39,22 +39,14 @@ def send_email_invitation(self, request, queryset):
             reset_form = PasswordResetForm({'email': user.email}, request.POST)
             assert reset_form.is_valid()
             reset_form.save(
-                #subject_template_name='templates/emails/account_creation_subject.txt',
-                #email_template_name='templates/emails/invitation_email.html',
+                subject_template_name='common_auth/emails/account_creation_subject.txt',
```
Comment (cwisecarver) [inline admin/common_auth/admin.py]: Neither of these appear to be there. Looks like the emails directory doesn't actually exist in common_auth unless I'm looking in the wrong root directory.

File: admin/common_auth/models.py [hunk 2]
```diff
@@ -39,14 +39,14 @@ class MyUser(AbstractBaseUser, PermissionsMixin):
...
     )
 
-    first_name = models.CharField(max_length=30, blank=True)
-    last_name = models.CharField(max_length=30, blank=True)
+    first_name = models.CharField(max_length=30, blank=False)
+    last_name = models.CharField(max_length=30, blank=False)
     is_active = models.BooleanField(default=True)
     is_admin = models.BooleanField(default=False)
     is_staff = models.BooleanField(default=False)
     date_joined = models.DateTimeField(default=datetime.now)
```
Comment (cwisecarver) [inline admin/common_auth/models.py]: Django has a way to [do this on add](https://docs.djangoproject.com/en/1.9/ref/models/fields/#django.db.models.DateField.auto_now_add). If you're migrating older data you'll have to decide what to do about the old ones.

File: admin/common_auth/models.py [hunk 3]
```diff
@@ -39,14 +39,14 @@ class MyUser(AbstractBaseUser, PermissionsMixin):
...
     confirmed = models.BooleanField(default=False)
-    osf_id = models.CharField(default=False, max_length=10, blank=True)
+    osf_id = models.CharField(default='', max_length=5, blank=True)
```
Comment (cwisecarver) [inline admin/common_auth/models.py]: I believe setting default and blank to True throws a warning.

File: admin/common_auth/views.py [hunk 4]
```diff
@@ -47,3 +55,45 @@ def get_success_url(self):
...
+
+
+class RegisterUser(SuperUser, FormView):
+    form_class = UserRegistrationForm
+    template_name = 'register.html'
+
+    def form_valid(self, form):
+        osf_id = form.cleaned_data.get('osf_id')
+        try:
+            osf_user = User.load(osf_id)
```
Comment (cwisecarver) [inline admin/common_auth/views.py]: Unless you expect `osf_user = User.load(osf_id)` to throw an AttributeError I'd pull it out of the try. It just limits the possibility you'll catch an exception that you weren't expecting.

File: admin/common_auth/views.py [hunk 5]
```diff
@@ -47,3 +55,45 @@ def get_success_url(self):
...
+            osf_user.system_tags.append(PREREG_ADMIN_TAG)
+        except AttributeError:
+            return page_not_found(
```
Comment (cwisecarver) [inline admin/common_auth/views.py]: Again, I'd raise `Http404('OSF user with id "{}" not found. Please double check.'.format(osf_id))` just for modularity.

**Comment by cwisecarver** [comment]
@Ghalko Clicking on "Forgot my password" in an incognito window gives me this:
screen shot 2016-04-26 at 12 18 15 pm

File: admin/base/urls.py [hunk 6]
```diff
@@ -10,21 +9,19 @@
...
+    url(
+        base_pattern,
         include([
             url(r'^$', views.home, name='home'),
-            url(r'^django_admin/', include(admin.site.urls)),
+            url(r'^admin/', include(admin.site.urls)),
             url(r'^spam/', include('admin.spam.urls', namespace='spam')),
-            url(r'^auth/', include('admin.common_auth.urls', namespace='auth')),
+            url(r'^account/', include('admin.common_auth.urls', namespace='auth')),
+            url(r'^password', include('password_reset.urls')),
```
Comment (cwisecarver) [inline admin/base/urls.py]: Think this needs a trailing slash.

File: admin/pre_reg/utils.py [hunk 7]
```diff
@@ -5,10 +5,14 @@
...
 
 def get_prereg_reviewers():
-    return MyUser.objects.filter(
-        groups__name='prereg_group'
-    ).annotate(
-        fuller_name=Concat('first_name', Value(' '), 'last_name')
-    ).values_list(
-        'email', 'fuller_name'
-    )
+    # Note - fixes django.db.utils.OperationalError: no such table' error if ever one erases a db table and tries to remigrate
```
Comment (cwisecarver) [inline admin/pre_reg/utils.py]: I think, if you moved the call for this from the form __init__ method to the form's get_initial method you wouldn't have this exception happen. Either way, the clause is too wide. You should trim it down to just OperationError. Lots of other things could go wrong in this statement and the wide exception handling would make it quite the mystery.

**Comment by cwisecarver** [comment]
Pass done. Back to you @Ghalko

File: admin/common_auth/admin.py (hunk 1 above)
Comment (Ghalko) [inline admin/common_auth/admin.py]: Changed strings to reflect actual files.

File: admin/common_auth/models.py (hunk 2 above)
Comment (Ghalko) [inline admin/common_auth/models.py]: Changed

File: admin/common_auth/models.py (hunk 3 above)
Comment (Ghalko) [inline admin/common_auth/models.py]: Removed default

File: admin/common_auth/views.py (hunk 4 above)
Comment (Ghalko) [inline admin/common_auth/views.py]: Right, done.

File: admin/common_auth/views.py (hunk 5 above)
Comment (Ghalko) [inline admin/common_auth/views.py]: Done.

File: admin/base/urls.py (hunk 6 above)
Comment (Ghalko) [inline admin/base/urls.py]: Fixed, seems to have fixed the error above too.

File: admin/pre_reg/utils.py (hunk 7 above)
Comment (Ghalko) [inline admin/pre_reg/utils.py]: I think this would be better addressed in #5478, and I would put it in the `FormView`'s `get_initial`.

**Comment by Ghalko** [comment]
@cwisecarver This is done.

**Comment by cwisecarver** [comment]
👍

**Comment by sloria** [comment]
Can you please update this with develop one last time, @Ghalko ?

**Comment by sloria** [comment]
Good to merge when the build passes.

**Pull request merged**